domain/relation/types.go

```diff
@@ -22,9 +23,18 @@ type GetRelationEndpointUUIDArgs struct {
 	RelationUUID corerelation.UUID
 }
 
+// RelationDetails represents the current application's view of a relation.
+type RelationDetails struct {
+	Life     life.Value
+	UUID     corerelation.UUID
+	ID       int
+	Key      string
```

Comment:
I know this is only scaffolding for the moment, but why both `ID` and `Key`? `Key` alone should be enough as a natural key right?

Reply:
Unfortunately the uniter uses both the natural key and a reaction ID, neither of which are a relation UUID.  The pattern of use is developing.  All three values are guaranteed unique in a model.